### PR TITLE
[Motions 2023 06 cwg 1] P2922R0 Core Language Working Group "ready" Issues

### LINKED ISSUE #6276: [2023-06 CWG Motion 1] P2922R0 Core Language Working Group "ready" Issues

### PR DESCRIPTION
**Pull request by burblebee**
Fixes #6276

Fixes cplusplus/papers#1465
Fixes cplusplus/papers#1467

Fixes cplusplus/CWG#237
Fixes cplusplus/CWG#278
Fixes cplusplus/CWG#279
Fixes cplusplus/CWG#280
Fixes cplusplus/CWG#291
Fixes cplusplus/CWG#292
Fixes cplusplus/CWG#293
Fixes cplusplus/CWG#294
Fixes cplusplus/CWG#296
Fixes cplusplus/CWG#297
Fixes cplusplus/CWG#300
Fixes cplusplus/CWG#301

### CODE REVIEW

**Review by JohelEGP — CHANGES_REQUESTED** [review]

File: source/declarations.tex [hunk 1]
```diff
@@ -7902,6 +7909,22 @@
...
+\begin{codeblock}
+struct C {
+  int i;
+};
+
+struct D1 : C { };
+struct D2 : C { };
+
+struct D3 : D1, D2 {
+  using D1::i;                  // OK, equivalent to using \tcode{C::i}
```
[the remark this answers is not in the document]
Comment (burblebee) [reply source/declarations.tex:7922]: thanks! fixed

File: source/lex.tex [hunk 2]
```diff
@@ -1317,21 +1317,27 @@
...
-A program is ill-formed
-if one of its translation units contains an \grammarterm{integer-literal}
-that cannot be represented by any of the allowed types.
+If an \grammarterm{integer-literal}
+cannot be represented by any of the allowed types,
+the program is ill-formed.
+\begin{note}
+An \grammarterm{integer-literal} with a \tcode{z} or \tcode{Z} suffix
+is ill-formed if it cannot be represented by \tcode{std::size_t}.
+\end{note}
```
[the remark this answers is not in the document]
Comment (burblebee) [reply source/lex.tex:1340]: Thanks! Resolving this for now.

File: source/lex.tex [hunk 3]
```diff
@@ -1317,21 +1317,27 @@
 \end{floattable}
 
 \pnum
-If an \grammarterm{integer-literal}
+Except for \grammarterm{integer-literal}{s} containing
```
[the remark this answers is not in the document]
Comment (burblebee) [reply source/lex.tex:1320]: Why? We typically encode Latex this way. Resolving - please reopen if you disagree.

[the remark this answers is not in the document]
Comment (JohelEGP) [reply source/lex.tex:1320]: The why is in the linked comment.

File: source/templates.tex [hunk 4]
```diff
@@ -2845,6 +2845,10 @@
 The instantiation of a \tcode{sizeof...} expression\iref{expr.sizeof} produces
 an integral constant with value $N$.
 
+\pnum
+The instantiation of an \grammarterm{alignment-specifier} with an ellipsis
+produces $\tcode{E}_1$, $\tcode{E}_2$, $\dotsc$, $\tcode{E}_N$.
```
[the remark this answers is not in the document]
Comment (burblebee) [reply source/templates.tex:2850]: Yes I did; reverted.

**Review by JohelEGP — APPROVED** [review]

**Pull request merged**